operators/src/processing/time_shift.rs

```diff
@@ -166,34 +166,70 @@ impl VectorOperator for TimeShift {
             (
                 RasterOrVectorOperator::Vector(source),
                 TimeShiftParams::Relative { granularity, value },
-            ) if value.is_positive() => Ok(Box::new(InitializedVectorTimeShift {
-                source: source.initialize(context).await?,
-                shift: RelativeForwardShift {
+            ) if value.is_positive() => {
+                let source = source.initialize(context).await?;
+
+                let shift = RelativeForwardShift {
                     step: TimeStep {
                         granularity,
                         step: value.unsigned_abs(),
                     },
-                },
-            })),
+                };
+
+                let mut result_descriptor = source.result_descriptor().clone();
+                if let Some(time) = result_descriptor.time {
+                    result_descriptor.time = shift.shift(time).map(|r| r.0).ok();
+                }
+
+                Ok(Box::new(InitializedVectorTimeShift {
+                    source,
+                    result_descriptor,
+                    shift,
+                }))
+            }
             (
                 RasterOrVectorOperator::Vector(source),
                 TimeShiftParams::Relative { granularity, value },
-            ) => Ok(Box::new(InitializedVectorTimeShift {
-                source: source.initialize(context).await?,
-                shift: RelativeBackwardShift {
+            ) => {
+                let source = source.initialize(context).await?;
+
+                let shift = RelativeBackwardShift {
                     step: TimeStep {
                         granularity,
                         step: value.unsigned_abs(),
                     },
-                },
-            })),
+                };
+
+                let mut result_descriptor = source.result_descriptor().clone();
+                if let Some(time) = result_descriptor.time {
+                    result_descriptor.time = shift.shift(time).map(|r| r.0).ok();
+                }
+
+                Ok(Box::new(InitializedVectorTimeShift {
+                    source,
+                    result_descriptor,
+                    shift,
+                }))
+            }
             (
                 RasterOrVectorOperator::Vector(source),
                 TimeShiftParams::Absolute { time_interval },
-            ) => Ok(Box::new(InitializedVectorTimeShift {
-                source: source.initialize(context).await?,
-                shift: AbsoluteShift { time_interval },
-            })),
+            ) => {
+                let source = source.initialize(context).await?;
+
+                let shift = AbsoluteShift { time_interval };
+
+                let mut result_descriptor = source.result_descriptor().clone();
+                if let Some(time) = result_descriptor.time {
+                    result_descriptor.time = shift.shift(time).map(|r| r.0).ok();
+                }
```

Comment:
It seems like you do this ~three~ six times. Could you extract a function?

Reply:
8d3e95f30fbd92e63850fadef41b8af083d3d461